driving_log_replayer/driving_log_replayer/perception_eval_conversions.py

```diff
@@ -232,6 +232,119 @@ def summarize_pass_fail_result(pass_fail: PassFailResult) -> dict:
     }
 
 
+def object_to_description(obj: ObjectType) -> dict:
+    return {
+        "label": obj.semantic_label.name,
+        "uuid": obj.uuid,
+        "position": obj.state.position,
+        "velocity": obj.state.velocity,
+        "orientation": obj.state.orientation.q.tolist(),  # quaternion to list
+    }
+
+
+def dynamic_object_result_to_error_description(obj: DynamicObjectWithPerceptionResult) -> dict:
+    pose_error = obj.distance_error_bev
+    heading_error = obj.heading_error
+    velocity_error = obj.velocity_error
+    return {
+        "pose_error": pose_error,
+        "heading_error": heading_error,
+        "velocity_error": velocity_error,
+    }
+
+
+def extract_pass_fail_objects_description(pass_fail: PassFailResult) -> list[dict]:
+    """
+    Extract detailed objects results from PassFailResult.
+
+    Args:
+    ----
+        pass_fail (PassFailResult): PassFailResult object
+
+    Returns:
+    -------
+        list[dict]: List of objects descriptions.
+        Each element is a dictionary with the following keys:
+            - "Objects"
+                - "status": "TP" | "FP" | "FN"
+                - "object_type": "GT" | "EST"
+                - "label": str
+                - "uuid": str
+                - "position": tuple[float, float, float]
+                - "velocity": tuple[float, float, float]
+                - "pose_error": optional[float]
+                - "heading_error": optional[float]
+                - "velocity_error": optional[float]
+                - "distance": float # distance between object center and ego vehicle
+                - "orientation": tuple[float, float, float, float]
+
+    """
+    ego2map_matrix = pass_fail.ego2map
+    has_map_to_base_link = len(ego2map_matrix) > 0
```

Comment:
ego2map is optional so none needs to be checked

```suggestion
    has_map_to_base_link = ego2map_matrix is not None and len(ego2map_matrix) > 0
```

Reply:
Thank you. I fixed this in https://github.com/tier4/driving_log_replayer/pull/452/commits/bd6eb695301f9679443fe4568debd5a128cce8f2.